meta-lmp-bsp/dynamic-layers/meta-ti-bsp/recipes-support/ti-mfgtool-files/ti-mfgtool-files/flash.sh.in

```diff
@@ -0,0 +1,190 @@
+
+# Validate we are running as root
+#   Future improvement could be that checks for udev rules to allow
+#   non-root running
+if [ "${EUID}" -ne "0" ]; then
+	error_exit "ERROR: Script not running as root"
+fi
+
+####
+# VALIDATE environment
+####
+
+# Validate required commands are available
+commands=("getopt" "dd" "udevadm" "dfu-util" "lsusb" "gunzip" "umount")
+validate_commands
+
+#######################
+# Parse Args
+#######################
+# NOTE: This requires GNU getopt.
+if ! PARSED_ARGS=$(getopt --longoptions "help,wic:,board-type:,dev-id:" \
+		--options "h" --name "${scriptName}" -- "$@") ; then
+	usage
+	error_exit "Unable to parse arguments - is GNU getopt installed?"
+fi
+
+# Note the quotes around '${PARSED_ARGS}' are essential
+eval set -- "${PARSED_ARGS}"
+
+# Make sure required option is defined
+if [ $# = 1 ]; then
+	usage
+	error_exit "Required option missing"
+fi
+
+# Board type
+boardType=
+# WIC filepath
+wicFile=
+# dev PID and VID
+devID=
+while true; do
+	case "$1" in
+		--wic) wicFile="${2}"; shift 2 ;;
+		--board-type) boardType="${2}"; shift 2 ;;
+		--dev-id) devID="${2}"; shift 2 ;;
+		# -- means the end of the arguments; drop this, and break out of the while loop
+		--) shift; break ;;
+		* ) usage; error_exit "Unrecognized option - ${1}" ;;
+	esac
+done
+
+# Ensure no positions args were specified.
+if [ $# != 0 ]; then
+	usage
+	error_exit "No positional arguments allowed - $*"
+fi
+
+#######################
+# Validate Parameters
+#######################
+if [ ! -f "${wicFile}" ]; then
+	error_exit "WIC image file not found!"
+fi
+
+[ -z "${boardType}" ] && boardType="gp"
+[ -z "${devID}" ] && devID="$(lsusb | grep 'Texas Instruments, Inc. AM62x' | awk '{print $6}')"
+
+####
+# Flash software images
+####
+
+# start USB DFU download mode and flash primary eMMC with provided WIC image
+
+echo "Load U-Boot via DFU..."
+
+echo "------------------------------------------"
+echo "  DFU BOOT TIBOOT3: TIFS and R5           "
+echo "------------------------------------------"
+dfu-util -R -a bootloader -D ${scriptDir}/tiboot3-am62x-${boardType}-evm.bin --device="${devID}" 2>&1 >/dev/null
+sleep 2
+# read -p "Boot tispl.bin" foo
+echo "------------------------------------------"
+echo "  DFU BOOT TISPL: TFA/OPTEE/ and A53 SPL  "
+echo "------------------------------------------"
+dfu-util -R -a tispl.bin -D ${scriptDir}/tispl.bin --device="${devID}" 2>&1 >/dev/null
+sleep 2
+# read -p "Boot u-boot.img " foo
+echo "------------------------------------------"
+echo "  DFU BOOT UBOOT: A53 UBOOT               "
+echo "------------------------------------------"
+dfu-util -R  -a u-boot.img -D ${scriptDir}/u-boot.img --device="${devID}" 2>&1 >/dev/null
+
+echo "------------------------------------------"
+echo "  Exposing eMMC via USB using UMS         "
+echo "------------------------------------------"
+
+sleep 6
+mmcPath=$(get_ums_disk)
+echo "Detected device: ${mmcPath}"
+
+if [ -z "${mmcPath}" ]
+then
```

Comment:
Some blocks with if / then in the same line, others in different lines, better to stick with one format here.

Reply:
fixed